packages/gravity/blade-runner/src/plan/env/replicant-env.ts

```diff
@@ -0,0 +1,151 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import { type Callback, Redis, type RedisOptions } from 'ioredis';
+
+import { Trigger } from '@dxos/async';
+import { Resource } from '@dxos/context';
+import { log } from '@dxos/log';
+import { isNode } from '@dxos/util';
+
+import { ReplicantRpcServer } from './replicant-rpc-server';
+import { REDIS_PORT, createRedisRpcPort } from './util';
+import { RESOURCE_USAGE_LOG, type ResourceUsageLogEntry } from '../../analysys/resource-usage';
+import { type ReplicantEnv } from '../interface';
+import { type ReplicantParams } from '../spec';
+
+export { type RedisOptions };
+
+export class ReplicantEnvImpl<S> extends Resource implements ReplicantEnv {
+  public redis: Redis;
```

Comment:
how is this different from the two below?

Reply:
@mykola-vrmchk add jsdoc pls